src/main/java/org/springframework/data/elasticsearch/core/convert/MappingElasticsearchConverter.java

```diff
@@ -474,6 +496,8 @@ public void write(Object source, Document sink) {
 
 		Assert.notNull(source, "source to map must not be null");
 
+		source = maybeCallbackBeforeConvert(source);
```

Comment:
Before convert callbacks have to go into the Template API to keep the Converter free from events and callbacks.

Reply:
I'll change that; I tought it nice to just have to add the callback calling to one place instead of the many ones where the mapping is triggered

Reply:
I agree that it makes sense to pul, things together. For MongoDB we introduced various types of operations utilities (`EntityOperations`, `QueryOperations`). The converter itself should not emit any events/callbacks as this kind of feature blurs the actual responsibility. 

On the flip side, MongoDB does this with its DBRef handling and we're not quite happy with its setup.